### DISCUSSION TITLE
Duplicated BBAN

### DISCUSSION THREAD

**Discussion by MildChaos** [firefly-iii/firefly-iii#7570] [General]
Hi,
Thank you and congratulation on a great piece of software! I started using Firefly III recently, and it's great. Well done!

I have one problem. My bank uses one BBAN number for all my debit card expenses. When importing csv file I just remove this bban number and all transactions are identified by creditor name correctly. It is working fine, except for additional work needed to clean the file. 
Lately I wanted to use Nordigen to automatically import data. However, all transactions identified by one BBAN even with different "creditorName" are clustered under one account and original "creditorName" is gone. This makes data unusable as I can't identity those expenses. 
Are there any chances to recover original creditor name, or maybe it could be possible to ignore bban during import and fetch data only from IBAN or creditor name?

Below, I pasted part of json file from nordigen. I removed sensitive data, but it should be readable.
`{
                "transactionId": "D202305310000009",
                "bookingDate": "2023-05-31",
                "valueDate": "2023-05-29",
                "transactionAmount": {
                    "amount": "-28.00",
                    "currency": "PLN"
                },
                "creditorName": "SPORTINFO WERONIKA KWIATE POZNAN",
                "creditorAccount": {
                    "bban": "10500031-1915031/19730"
                },
                "debtorName": "S*** B***",
                "debtorAccount": {
                    "iban": "PL7610*******************"
                },
                "remittanceInformationUnstructured": "Płatność kartą 29.05.2023 Nr karty 4246xx4771",
                "remittanceInformationUnstructuredArray": [
                    "Płatność kartą 29.05.2023 Nr karty 4246xx4771"
                ],
                "balanceAfterTransaction": {
                    "balanceAmount": {
                        "amount": "***.**",
                        "currency": "PLN"
                    },
                    "balanceType": "interimBooked"
                },
                "internalTransactionId": "d1c6acd9aa3d3a0b67149bb232efab79"
            },
            {
                "transactionId": "D202305310000007",
                "bookingDate": "2023-05-31",
                "valueDate": "2023-05-29",
                "transactionAmount": {
                    "amount": "-150.00",
                    "currency": "PLN"
                },
                "creditorName": "SOFA KOSMETYKA I FRYZJERS POZNAN",
                "creditorAccount": {
                    "bban": "10500031-1915031/19730"
                },
                "debtorName": "S*** B***",
                "debtorAccount": {
                    "iban": "PL7610*******************"
                },
                "remittanceInformationUnstructured": "Płatność kartą 29.05.2023 Nr karty 4246xx4771",
                "remittanceInformationUnstructuredArray": [
                    "Płatność kartą 29.05.2023 Nr karty 4246xx4771"
                ],
                "balanceAfterTransaction": {
                    "balanceAmount": {
                        "amount": "***.**",
                        "currency": "PLN"
                    },
                    "balanceType": "interimBooked"
                },
                "internalTransactionId": "69d19896e07ea8942ee2fc6b629c32cd"
            },
            {
                "transactionId": "D202305310000005",
                "bookingDate": "2023-05-31",
                "valueDate": "2023-05-29",
                "transactionAmount": {
                    "amount": "-70.81",
                    "currency": "PLN"
                },
                "creditorName": "JMP S.A. BIEDRONKA 4511   BIEDRUSKO",
                "creditorAccount": {
                    "bban": "10500031-1915031/19730"
                },
                "debtorName": "S*** B***",
                "debtorAccount": {
                    "iban": "PL7610*******************"
                },
                "remittanceInformationUnstructured": "Płatność kartą 29.05.2023 Nr karty 4246xx4771",
                "remittanceInformationUnstructuredArray": [
                    "Płatność kartą 29.05.2023 Nr karty 4246xx4771"
                ],
                "balanceAfterTransaction": {
                    "balanceAmount": {
                        "amount": "***.**",
                        "currency": "PLN"
                    },
                    "balanceType": "interimBooked"
                },
                "internalTransactionId": "7e8b1020cb882cdbc8d615ddfa2a2a13"
            },
            {
                "transactionId": "D202305290000007",
                "bookingDate": "2023-05-29",
                "valueDate": "2023-05-29",
                "transactionAmount": {
                    "amount": "-2.06",
                    "currency": "PLN"
                },
                "creditorName": "S*** Z***",
                "creditorAccount": {
                    "iban": "PL4210*******************"
                },
                "debtorName": "S*** B***",
                "debtorAccount": {
                    "iban": "PL7610*******************"
                },
                "remittanceInformationUnstructured": "przelew Kwota: *** PLN",
                "remittanceInformationUnstructuredArray": [
                    "przelew  Kwota: *** PLN"
                ],
                "balanceAfterTransaction": {
                    "balanceAmount": {
                        "amount": "***.**",
                        "currency": "PLN"
                    },
                    "balanceType": "interimBooked"
                },
                "internalTransactionId": "af62af14d3b422a46cf6f6d54b19939d"
            },`

Thanks in advance.
Best regards,
Bogusz

**Comment by JC5** [comment]
You can try to patch this using rules, but ultimately it's your bank that delivers the data.
